### PR TITLE
Add AuthStatus type

### PR DESCRIPTION
**Pull request by karthikmurali60**
# Changes

- :gift: Add new feature
- This PR adds a new AuthStatus type which currently contains the ServiceAccountName attribute, which is used for OIDC authentication

/kind api-change

Fixes https://github.com/knative/eventing/issues/7173

**Release Note**

```release-note
Add AuthStatus type
```

**Docs**

```docs

```

### CODE REVIEW

**Comment by karthikmurali60** [comment]
@pierDipi can you please review this PR?

**Review by pierDipi — COMMENTED** [review]
Thanks @karthikmurali60!

/approve

/cc @creydr

File: apis/duck/v1/auth_types.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+// +genduck
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
```
Comment (creydr) [inline apis/duck/v1/auth_types.go:21]:
```suggestion
```
I think we don't need this

**Review by creydr — CHANGES_REQUESTED** [review]

**Comment by creydr** [comment]
@karthikmurali60 you have to run `hack/update-codegen.sh` as you updated the types

**Comment by karthikmurali60** [comment]
@creydr I ran the script and committed the generated files. Still some of the tests are failing, this is because of the missing functions in `auth_types.go` file. I am trying to add them

**Comment by creydr** [comment]
Hey @karthikmurali60,
seems the generated file was not updated in your [latest push](https://github.com/knative/pkg/compare/f71397e06238450b4203dda2b77a5977767291a3..999ce35e0d1e401d57c84d36815852f0ad373bbb). 
When I run the `hack/update-codegen.sh` on your branch I see some changes. So I think running it and pushing all the files should fix the build issues.

**Comment by karthikmurali60** [comment]
@creydr I ran `hack/update-codegen.sh` on this branch and pushed the changes in [this](https://github.com/knative/pkg/pull/2829/commits/5c0b353b9e538c1e51100b51be32c6723203b1e9) commit. However, the same tests are still failing. I am not able to find out why this is happening. Any thoughts ?

**Comment by dprotaso** [comment]
Error states
```
=== Errors
client/injection/ducks/duck/v1/authstatus/authstatus.go:44:37: (&v1.AuthStatus{}).GetFullType undefined (type *"knative.dev/pkg/apis/duck/v1".AuthStatus has no field or method GetFullType)
```

**Comment by creydr** [comment]
> Error states
> 
> ```
> === Errors
> client/injection/ducks/duck/v1/authstatus/authstatus.go:44:37: (&v1.AuthStatus{}).GetFullType undefined (type *"knative.dev/pkg/apis/duck/v1".AuthStatus has no field or method GetFullType)
> ```

@karthikmurali60 It seems this issue is because you had the `+genduck` annotation, when you run update-codegen.sh initially (and it seems update-codegen does not update client/injection/... :shrug: ). 
I would remove the generated files from `client/injection/ducks/duck/v1/authstatus` and rerun update-codegen.sh.

**Comment by karthikmurali60** [comment]
@creydr this was the issue, thanks a lot for pointing that out. I have fixed it and pushed the new changes. 

> > Error states
> > ```
> > === Errors
> > client/injection/ducks/duck/v1/authstatus/authstatus.go:44:37: (&v1.AuthStatus{}).GetFullType undefined (type *"knative.dev/pkg/apis/duck/v1".AuthStatus has no field or method GetFullType)
> > ```
> 
> @karthikmurali60 It seems this issue is because you had the `+genduck` annotation, when you run update-codegen.sh initially (and it seems update-codegen does not update client/injection/... 🤷 ). I would remove the generated files from `client/injection/ducks/duck/v1/authstatus` and rerun update-codegen.sh.

**Comment by creydr** [comment]
> @creydr this was the issue, thanks a lot for pointing that out. I have fixed it and pushed the new changes.

Awesome :tada: Lets wait for the other checks to pass and then we should be ready to go :+1:

File: apis/duck/v1/auth_types.go
Comment (creydr) [inline apis/duck/v1/auth_types.go:1]: @karthikmurali60 seems the boilerplate does not match the template (some missing lines, etc.). You can check on https://github.com/knative/pkg/blob/main/hack/boilerplate/boilerplate.go.txt for the format (just replace the year)

**Pull request merged**